renderdoc/driver/gl/gl_replay.cpp

```diff
@@ -2236,103 +2236,117 @@ byte *GLReplay::GetTextureData(ResourceId tex, uint32_t arrayIdx, uint32_t mip,
     arraysize = texDetails.depth;
   }
 
-  if(params.remap && intFormat != eGL_RGBA8 && intFormat != eGL_SRGB8_ALPHA8)
+  if(params.remap)
   {
-    RDCASSERT(params.remap == eRemap_RGBA8);
+    GLenum remapFormat = eGL_RGBA8;
+    if(params.remap == eRemap_RGBA8)
+      remapFormat = eGL_RGBA8;
+    else if(params.remap == eRemap_RGBA16)
+      remapFormat = eGL_RGBA16;
+    else if(params.remap == eRemap_RGBA32)
+    {
+      if(!IsGLES || HasExt[EXT_color_buffer_float])
+        remapFormat = eGL_RGBA32F;
+      else
+        remapFormat = eGL_RGBA32UI;
```

Comment:
I don't exactly understand what you mean that the texture display rendering only supports rendering out to floating point targets; I thought the value written out by the fragment shader won't get clamped if the rendertarget is an integer format, so the shader for the texture copy should be identical whether the rendertarget is floating point or integer.

If you mean the ranges between 0 and 1 will be lost, should I perhaps add another remap mode eRemap_RGBA32UI ? GL_DEPTH_COMPONENT24 is actually an integer format so remapping to another integer format might be more approriate.